### PR TITLE
Add documentation and tests for `TorchLayer` saving

### PR DESCRIPTION
**Pull request by eddddddy**
**Context:**
`TorchLayer` is already able to be saved using the usual PyTorch syntax.

**Description of the Change:**
Add documentation on how to do it. Also add tests in case it breaks in the future.

### CODE REVIEW

**Comment by eddddddy** [comment]
I don't think this change requires a changelog entry.

File: pennylane/qnn/torch.py [hunk 1]
```diff
@@ -150,6 +150,30 @@ def qnode(inputs, weights_0, weights_1, weights_2, weight_3, weight_4):
...
             qlayer = qml.qnn.TorchLayer(qnode, weight_shapes=weight_shapes, init_method=init_method)
 
+        **Model saving**
+
+        Instances of ``TorchLayer`` can be saved using the usual ``torch.save()`` utility:
+
+        .. code-block::
+
+            qlayer = qml.qnn.TorchLayer(qnode, weight_shapes=weight_shapes)
+            torch.save(qlayer.state_dict(), SAVE_PATH)
```
Comment (trbromley) [inline pennylane/qnn/torch.py:160]: @eddddddy, could we have an example that includes a model with multiple layers? E.g.
```suggestion
            qlayer = qml.qnn.TorchLayer(qnode, weight_shapes=weight_shapes)
            clayer = torch.nn.Linear(2, 2)
            model = torch.nn.Sequential(clayer, qlayer)
            torch.save(model.state_dict(), SAVE_PATH)
```

File: pennylane/qnn/torch.py [hunk 2]
```diff
@@ -150,6 +150,30 @@ def qnode(inputs, weights_0, weights_1, weights_2, weight_3, weight_4):
...
+
+        To load the model, an instance of the class must be created first before calling ``torch.load()``,
+        as required by PyTorch:
+
+        .. code-block::
+
+            qlayer = qml.qnn.TorchLayer(qnode, weight_shapes=weight_shapes)
+            qlayer.load_state_dict(torch.load(SAVE_PATH))
+            qlayer.eval()
```
Comment (trbromley) [inline pennylane/qnn/torch.py:169]:
```suggestion
            qlayer = qml.qnn.TorchLayer(qnode, weight_shapes=weight_shapes)
            clayer = torch.nn.Linear(2, 2)
            model = torch.nn.Sequential(clayer, qlayer)
            model.load_state_dict(torch.load(SAVE_PATH))
            model.eval()
```
For the comment above.

File: pennylane/qnn/torch.py (hunk 1 above)
Comment (trbromley) [reply pennylane/qnn/torch.py:160]: And maybe we have this in the tests?

**Review by rmoyard — APPROVED** [review]
Thanks that's great and easier than expected! I agree with @trbromley comment on adding a clayer in the doc otherwise it looks good to me :+1:

File: pennylane/qnn/torch.py (hunk 1 above)
Comment (eddddddy) [reply pennylane/qnn/torch.py:160]: Added another example in the docstring on top of the previous one. Also changed the test.

**Review by trbromley — APPROVED** [review]
Thanks @eddddddy!

**Comment by eddddddy** [comment]
[sc-36427]

**Pull request merged**